packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -719,323 +644,136 @@ class TimeInput extends React.Component {
       ...customProps
     } = this.props;
 
-    const instanceHoursAttrs = { ...hourAttributes };
-    const instanceMinuteAttrs = { ...minuteAttributes };
-    const instanceSecondAttrs = { ...secondAttributes };
-
     // Using the state of hour, minute, and second (if shown) create a time in UTC represented in ISO 8601 format.
     let timeValue = '';
 
-    if (this.state.hour.length > 0 || this.state.minute.length > 0 || this.state.second.length > 0) {
+    if (this.state.hour.length > 0 || this.state.minute.length > 0 || (this.state.second.length > 0 && showSeconds)) {
       let hour = parseInt(this.state.hour, 10);
 
-      if (this.props.variant === TimeUtil.FORMAT_12_HOUR && this.state.meridiem === this.postMeridiem && hour > 12) {
+      if (this.props.variant === TimeUtil.FORMAT_12_HOUR && this.state.meridiem === this.postMeridiem) {
         hour += 12;
       }
 
-      timeValue = 'T'.concat(hour, ':', this.state.minute).concat(showSeconds ? ':'.concat(this.state.second) : '');
-    }
-
-    if (!instanceHoursAttrs.id) {
-      instanceHoursAttrs.id = 'terra-time-hour-'.concat(name);
-    }
+      timeValue = 'T'.concat(hour, ':', this.state.minute);
 
-    if (!instanceMinuteAttrs.id) {
-      instanceMinuteAttrs.id = 'terra-time-minute-'.concat(name);
+      if (showSeconds) {
+        timeValue = timeValue.concat(':', this.state.second);
+      }
     }
 
-    if (!instanceSecondAttrs.id) {
-      instanceSecondAttrs.id = 'terra-time-second-'.concat(name);
-    }
+    const timeInputClassNames = cx([
+      { disabled },
+      'time-input',
+      { 'is-focused': this.state.isFocused },
+      customProps.className,
+    ]);
 
+    /* eslint-disable jsx-a11y/no-static-element-interactions */
     return (
       <div
         {...customProps}
-        className={cx(['mobile-time-picker', customProps.className])}
         ref={this.timeInputContainer}
+        className={cx('time-input-container')}
       >
-        <input
-          // Create a hidden input for storing the name and value attributes to use when submitting the form.
-          // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
-          type="hidden"
-          name={name}
-          value={timeValue}
-        />
-        <div className={cx('time-input-group')}>
+        <div className={timeInputClassNames}>
+          <input
+            // Create a hidden input for storing the name and value attributes to use when submitting the form.
+            // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
+            type="hidden"
+            name={name}
+            value={timeValue}
+          />
           <Input
             {...inputAttributes}
-            {...instanceHoursAttrs}
+            {...minuteAttributes}
```

Comment:
This should be hourAttributes I believe.
```suggestion
            {...hourAttributes}
```